### ISSUE TITLE
parsing datatypes in LH

### ISSUE THREAD

**Issue by zilinc** [ucsd-progsys/liquidhaskell#873]
When defining the following:
```
{-@ type TypeName = String @-}
type TypeName = String

{-@
data Type <p :: Int -> Prop>
  = TVar (v :: Int <p v>)
  | TCon (TypeName) [Type <p>]  -- TypeName has to be within parens
@-}

data Type
  = TVar Int
  | TCon TypeName [Type]
```
two parser problems occur:
1. as the comment suggests
2. the comment itself is not parsed

**Comment by kosmikus** [comment]
I'd argue this is fixed with the upcoming parser changes, and probably related to #871.

We have to look at various aspects independently:

- `Prop` has to become `Bool`. I think this is backed up by other changes made to LH in the meantime.
- The `(TypeName)` parens can indeed be removed now.
- The comment has to be introduced with `//`, because for some reason, that is the line comment syntax that LH defines. We might want to independently change this and adapt it to be more Haskell-like.

**Comment by ranjitjhala** [comment]
Great, shall we close or perhaps wait till the changes are merged? Thanks!!!

**Comment by kosmikus** [comment]
Fixed via #1704
